Explorer/Assets/DCL/Multiplayer/Movement/Systems/RemotePlayersMovementSystem.cs

```diff
@@ -0,0 +1,243 @@
+﻿using Arch.Core;
+using Arch.System;
+using Arch.SystemGroups;
+using Arch.SystemGroups.DefaultSystemGroups;
+using DCL.AvatarRendering.AvatarShape.UnityInterface;
+using DCL.Character.Components;
+using DCL.CharacterMotion.Animation;
+using DCL.CharacterMotion.Components;
+using DCL.Diagnostics;
+using DCL.Multiplayer.Connections.RoomHubs;
+using DCL.Multiplayer.Movement.Settings;
+using DCL.Multiplayer.Movement.System;
+using ECS.Abstract;
+using UnityEngine;
+using Utility.PriorityQueue;
+
+namespace DCL.Multiplayer.Movement.Systems
+{
+    [UpdateInGroup(typeof(PresentationSystemGroup))]
+    [LogCategory(ReportCategory.MULTIPLAYER_MOVEMENT)]
+    public partial class RemotePlayersMovementSystem : BaseUnityLoopSystem
+    {
+        private readonly IMultiplayerMovementSettings settings;
+        private readonly RemotePlayersMovementInbox messagePipe;
+
+        public RemotePlayersMovementSystem(World world, IRoomHub roomHub, IMultiplayerMovementSettings settings) : base(world)
+        {
+            this.settings = settings;
+            messagePipe = new RemotePlayersMovementInbox(roomHub, settings);
+        }
+
+        protected override void Update(float t)
+        {
+            UpdateRemotePlayersMovementQuery(World, t);
+        }
+
+        private static void HandleFirstMessage(FullMovementMessage firstRemote, ref CharacterTransform transComp, ref CharacterAnimationComponent anim, ref RemotePlayerMovementComponent remotePlayerMovement, in IAvatarView view)
+        {
+            transComp.Transform.position = firstRemote.position;
+            UpdateAnimations(firstRemote, ref anim, view);
+
+            remotePlayerMovement.AddPassed(firstRemote, wasTeleported: true);
+            remotePlayerMovement.Initialized = true;
+        }
+
+        [Query]
+        [None(typeof(PlayerComponent))]
+        private void UpdateRemotePlayersMovement([Data] float deltaTime, ref CharacterTransform transComp, ref CharacterAnimationComponent anim,
```

Comment:
Can we disconnect animation state from the movement interpolation as this function has too many responsibilities?

Reply:
As we agreed, let's handle this separation on a separate branch after the merge